Tests/SourceKitLSPTests/RangeFormattingTests.swift

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SKLogging
+import SKTestSupport
+import SourceKitLSP
+import XCTest
+
+final class RangeFormattingTests: XCTestCase {
+  func testSelectComment() async throws {
+    try await SkipUnless.toolchainContainsSwiftFormat()
+    let testClient = try await TestSourceKitLSPClient()
+    let uri = DocumentURI(for: .swift)
+
+    let positions = testClient.openDocument(
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      1️⃣// do stuff2️⃣
+      }
+      }
+      """,
+      uri: uri
+    )
+
+    let response = try await testClient.send(
+      DocumentRangeFormattingRequest(
+        textDocument: TextDocumentIdentifier(uri),
+        range: positions["1️⃣"]..<positions["2️⃣"],
+        options: FormattingOptions(tabSize: 2, insertSpaces: true)
+      )
+    )
+
+    let edits = try XCTUnwrap(response)
+    XCTAssertEqual(
+      edits,
+      [
+        TextEdit(range: positions["1️⃣"]..<positions["2️⃣"], newText: "    // do stuff")
+      ]
+    )
```

Comment:
@ahoppen When I trigger this test, `edits` turns out to be empty no matter where I put the markers.
Any reasons? maybe I missed something obvious?
I don't notice any errors in the logs.

Reply:
I think this is because of the line vs. UTF-8 offset issue I mentioned above.

Reply:
Thank you.